packages/react-tabs/src/components/Tab/useTabAnimatedIndicator.ts

```diff
@@ -0,0 +1,185 @@
+        `${pendingAnimationDurationTokens.slow}`,
+      transitionTimingFunction:
+        `${pendingAnimationEasingTokens.declerateMax},` +
+        `${pendingAnimationEasingTokens.declerateMax},` +
+        `${pendingAnimationEasingTokens.declerateMax}`,
+    },
+    '@media (prefers-reduced-motion: reduce)': {
+      ':after': {
+        transitionProperty: 'none',
+        transitionDuration: '0ms',
+      },
+    },
+  },
+  mediumHorizontal: {
+    ':after': {
+      left: `calc(var(${indicatorCssVars.startOffsetVar}) + ${tabIndicatorPadding.mediumHorizontal})`,
+      right: `calc(var(${indicatorCssVars.endOffsetVar}) + ${tabIndicatorPadding.mediumHorizontal})`,
+    },
+  },
+  mediumVertical: {
+    ':after': {
+      top: `calc(var(${indicatorCssVars.startOffsetVar}) + ${tabIndicatorPadding.mediumVertical})`,
+      bottom: `calc(var(${indicatorCssVars.endOffsetVar}) + ${tabIndicatorPadding.mediumVertical})`,
+    },
+  },
+  smallHorizontal: {
+    ':after': {
+      left: `calc(var(${indicatorCssVars.startOffsetVar}) + ${tabIndicatorPadding.smallHorizontal})`,
+      right: `calc(var(${indicatorCssVars.endOffsetVar}) + ${tabIndicatorPadding.smallHorizontal})`,
+    },
+  },
+  smallVertical: {
+    ':after': {
+      top: `calc(var(${indicatorCssVars.startOffsetVar}) + ${tabIndicatorPadding.smallVertical})`,
+      bottom: `calc(var(${indicatorCssVars.endOffsetVar}) + ${tabIndicatorPadding.smallVertical})`,
+    },
+  },
+});
+
+const calculateTabRect = (element: HTMLElement) => {
+  if (element) {
+    const parentRect = element.parentElement?.getBoundingClientRect() || { x: 0, y: 0, width: 0, height: 0 };
+    const tabRect = element.getBoundingClientRect();
+
+    return {
+      x: tabRect.x - parentRect.x,
+      y: tabRect.y - parentRect.y,
+      width: tabRect.width,
+      height: tabRect.height,
+    };
+  }
+  return undefined;
+};
+
+/**
+ * Adds additional styling to the active tab selection indicator to create a sliding animation.
+ */
+export const useTabAnimatedIndicatorStyles_unstable = (state: TabState): TabState => {
+  const activeIndicatorStyles = useActiveIndicatorStyles();
+
+  const { disabled, selectedValue, value, vertical } = state;
+  const selected = selectedValue === value;
+
+  const { previousSelectedValue, registeredTabs } = useContextSelector(TabListContext, ctx => ({
+    previousSelectedValue: ctx.previousSelectedValue,
+    registeredTabs: ctx.registeredTabs,
+  }));
+
+  const previousSelectedTabRect = React.useMemo(() => {
+    const element = registeredTabs[JSON.stringify(previousSelectedValue)]?.ref.current;
+    return element ? calculateTabRect(element) : undefined;
+  }, [previousSelectedValue, registeredTabs]);
+
+  const selectedTabRect = React.useMemo(() => {
+    const element = registeredTabs[JSON.stringify(selectedValue)]?.ref.current;
+    return element ? calculateTabRect(element) : undefined;
+  }, [selectedValue, registeredTabs]);
```

Comment:
Is it false positive during initial render? As during first render `element` will always `null` (refs are resolved after rendering)

Reply:
Right - the registration happens after the initial render so it shouldn't false positive and the rect should be undefined.  Maybe I don't understand - do you mean by false positive that there would be a calculated rect during first render?

Reply:
> Maybe I don't understand - do you mean by false positive that there would be a calculated rect during first render?

Correct, may be I understood it wrong, but code relies on `selectedTabRect`. I meant that it works for updates (renders after initial one), but after initial render the initial state might be wrong.

Reply:
When the selectedTabRect is undefined, then the offsets will not be set and will be zero. This will cause the active selection indicator to be placed under the tab if it is selected. So for the initial render if the tab is selected, the indicator will appear correctly. The only way for the selection to change post render is by selecting a tab which would have to be mounted and it should then have a rectangle.